common/kvstore/src/test/java/org/apache/spark/util/kvstore/RocksDBSuite.java

```diff
@@ -381,6 +382,55 @@ public void testSkipAfterDBClose() throws Exception {
     assertFalse(iter.skip(1));
   }
 
+  @Test
+  public void testResourceCleaner() throws Exception {
+    File dbPathForCleanerTest = File.createTempFile(
+      "test_db_cleaner.", ".rdb");
+    dbPathForCleanerTest.delete();
+
+    RocksDB dbForCleanerTest = new RocksDB(dbPathForCleanerTest);
+    try {
+      for (int i = 0; i < 8192; i++) {
+        dbForCleanerTest.write(createCustomType1(i));
+      }
+      RocksDBIterator<CustomType1> rocksDBIterator =
+        (RocksDBIterator<CustomType1>) dbForCleanerTest.view(CustomType1.class).iterator();
+      Reference<RocksDBIterator<?>> reference =
```

Comment:
In this case, since we didn't manually close the `RocksDBIterator`, we can ultimately determine whether the `RocksDBIterator` is closed through `resourceCleaner.getStatus()`. 

So here, `reference` is just an auxiliary tool to judge whether the `RocksDBIterator` has been GCed. So can we change it to manually wrap a `WeakReference` instead of getting it from the `iteratorTracker`, so that we don't need to add an `Accessor` for the `iteratorTracker`?